### PR TITLE
#205 add fail-safe mode

### PR DESCRIPTION
**Pull request by Almenon**
see #205

### CODE REVIEW

**Comment by Almenon** [comment]
github's resolve conflicts is timing out, not sure why :(

File: jsonpickle/pickler.py [hunk 1]
```diff
@@ -85,13 +107,17 @@ def __init__(self,
...
+                 use_decimal=False,
+                 use_base85=False,
+                 fail_safe=None):
         self.unpicklable = unpicklable
         self.make_refs = make_refs
         self.backend = backend or json
         self.keys = keys
         self.warn = warn
         self.numeric_keys = numeric_keys
+        self.use_base85 = use_base85 and (not PY2)
```
Comment (davvid) [inline jsonpickle/pickler.py]: The parens here are superfluous.

File: jsonpickle/pickler.py [hunk 2]
```diff
@@ -584,4 +649,4 @@ def _wrap_string_slot(string):
     """
     if isinstance(string, string_types):
         return (string,)
-    return string
+    return string
```
Comment (davvid) [inline jsonpickle/pickler.py]: New newline at end of file?  What editor are you using?  Please adjust your settings.

Comment (Almenon) [reply jsonpickle/pickler.py]: I use VSCode. Not sure why that happened, I don't have it set to do that as far as I am aware.

File: jsonpickle/pickler.py (hunk 1 above)
Comment (Almenon) [reply jsonpickle/pickler.py]: I didn't make that change lol. Github's compare view was wonky. I just updated it and now it has a better view.

**Comment by Almenon** [comment]
> It looks like you merged master into your branch. Please rebase against our master branch, don't merge our branch into yours. Sorry for being a stickler -- but commits like "update pickler to match repo" should not exist in the history at all since it makes it non-bisectable, i.e. your branch history is all intermingled with changes from master that are unrelated to your changes, which makes them hard to review and test.

> Once you rebase those commits away you should be left with just your changes, and nothing else.

I think I used github's merge conflict editor to fix the merge conflicts, and that was what did the merge. 

I went back in history to before the merge, reset my repo to the upstream head, and rebased my commits on top of master.  I'll create another PR with better history.

> Regarding VS code, please see:
microsoft/vscode#12076 (comment)

> Please configure your editor accordingly. Its defaults are crummy.

Done

**Pull request closed without merging**